2 - RTB/Documentazione esterna/Analisi dei Requisiti/Analisi dei Requisiti.typ

```diff
@@ -237,26 +245,36 @@ $bold("Scenario: ")$
 
 
 
-== UC-4 Creazione scaffale
+= Richiesta di spostamento di un prodotto
+
+#figure(image("./imgs/uc4.png", format: "png"), caption: [UML UC-4])
+
+== Richiesta di spostamento di un prodotto
+
 $bold("Descrizione: ")$ 
-uno scaffale viene creato in base ai valori dati dall'utente e aggiunto nell'ambiente in una posizione valida specificata. Seccessivamente vengono creati i bin contenuti dallo scaffale e posizionati in esso.
+l'utente seleziona il prodotto di cui desidera una ricollocazione all'interno del magazzino e avvia una richiesta di spostamento verso un altro bin.
 
 $bold("Attore: ")$
 utente.
 
 $bold("Precondizioni: ")$
-- l'ambiente di lavoro deve essere stato configurato con successo.
+- devono esistere almeno due bin distinti;
+- uno dei due bin deve contenere un prodotto;
+- uno dei due bin deve essere vuoto.
 
 $bold("Postcondizioni: ")$
-- nell'ambiente di lavoro è stato aggiunto un nuovo scaffale;
-- nello scaffale creato sono stati aggiunti i bin da esso contenuti.
+- il prodotto è contenuto all'interno del bin di destinazione;
+- il bin di partenza risulta vuoto;
+- viene inviata una richiesta di spostamento al magazzino tramite l'uso di API;
+- il bin di partenza viene colorato in modo da identificare il fatto che da quel bin è in atto uno spostamento;
```

Comment:
il fatto di "venire colorato" rispecchia una scelta di design: direi un generico "il bin di partenza viene evidenziato (che poi sia tramite colore, tramite animazione, trami qualsiasi cosa, è indifferente) in modo da riconoscere che è stata inoltrata la richiesta di spostamento"